src/core/tranche/distributor.sol

```diff
@@ -0,0 +1,170 @@
+        t.operator = OperatorLike(operator_);
+        tranches.push(t);
+    }
+
+    function ratioOf(uint i) public auth returns (uint) {
+        return tranches[i].ratio/100;
+    }
+
+    // this function assumes two tranches only
+    function equityRatio() public auth returns (uint) {
+        return tranches[0].ratio;
+    }
+
+    // if capital should flow through, all funds in reserve should be moved in pile
+    function tradFlow(bool poolClosing) public auth {
+        if (!poolClosing) {
+            for (uint i = 0; i < tranches.length; i++) {
+                // calculates how much money is in the reserve, transfers all of this balance to the pile
+                uint wadR = tranches[i].operator.balance();
+                tranches[i].operator.borrow(address(pile), uint(wadR));
+                }
+        } else {
+            // payout
+            waterfallRepay();
+        }
+    }
+
+    function customFlow(bool poolClosing) public auth {
+        if (!poolClosing) {
+            poolOpen();
+        } else {
+            // this should take the money from the pile, repay and close the vault
+            waterfallRepay();
+        }
+    }
+
+    function poolOpen() private {
+        int wad = pile.want();
+        // pile wants money
+        if (wad > 0) {
+            pileWants();
+        } else if (wad < 0) {
+            pileHas();
+        }
+    }
+
+    function pileWants() private {
+        int wad = pile.want();
+        // should always take money from the equity tranche first
+        for (uint i = 0; i < tranches.length; i++) {
+            uint wadR = tranches[i].operator.balance();
+            // this can be cast into uint bc wad should always be positive at this point
+            if (uint(wad) < wadR) {
+                tranches[i].operator.borrow(address(pile), uint(wad));
+                wad = 0;
+            } else {
+                tranches[i].operator.borrow(address(pile), uint(wadR));
+                wad = int(uint(wad) - wadR);
+            }
+        }
+    }
+
+    function pileHas() private {
+        int wad = pile.want();
+        // pile has extra money, this can be repaid into the reserve
+        // senior tranche should always be the last tranche in the array and be repaid first
+        for (int i = int(tranches.length - 1); i > int(-1); i--) {
+            QuantLike quant = tranches[uint(i)].operator.quant();
+            int wadD = int(quant.debt());
+            // positive number here means there is some debt in the senior tranche
+            if (wad < int(wadD*-1) && wadD > 0) {
+                // wadD cast to uint bc it should be positive here
+                tranches[uint(i)].operator.repay(address(pile), uint(wadD));
+                wad = wad + wadD;
+            } else {
+                // wad cast to uint bc it should be positive here
+                tranches[uint(i)].operator.repay(address(pile), uint(wad));
+                wad = 0;
+            }
```

Comment:
I think the code can be simplified to:
```javascript
if (wadD >= wad) {
 tranches[uint(i)].operator.repay(address(pile), uint(wad));
 return;
}
 tranches[uint(i)].operator.repay(address(pile), uint(wadD));
 wad = sub(wad, wadD);
```

Reply:
taking this suggestion into account but flipping the negatives/positives bc wad here is a negative number (pileHas)